File: Dockerfile.java
# Stage 1: builder
FROM maven:3.9-eclipse-temurin-21 as builder

WORKDIR /app

COPY src/java/pom.xml .
RUN mvn dependency:go-offline -B

COPY src/java .
RUN mvn package -DskipTests -B

# Stage 2: runtime
FROM eclipse-temurin:21-jre-alpine

WORKDIR /app

COPY --from=builder /app/target/*.jar app.jar

EXPOSE 8081

ENTRYPOINT ["java", "-jar", "app.jar"]

File: Dockerfile.go
# Stage 1: builder
FROM golang:1.21-alpine as builder

WORKDIR /app

COPY src/go/go.mod ./
RUN go mod download
COPY src/go .
RUN CGO_ENABLED=0 go build -o /server ./cmd/server

# Stage 2: runtime
FROM alpine:3.19

WORKDIR /app

COPY --from=builder /server .

EXPOSE 8080

ENTRYPOINT ["./server"]
